### PR TITLE
More visibility spec implementation.

### PR DESCRIPTION
**Pull request by avimehta**
This PR adds support for:
Conditions:
    "continuousTimeMin": <milliseconds>,
    "continuousTimeMax": <milliseconds>,
    "totalTimeMin": <milliseconds>,
    "totalTimeMax": <milliseconds>,

And Macros:
    `minVisiblePercentage`
    `maxVisiblePercentage`
    `totalVisibleTime`
    `continuousVisibleTime`
    `firstSeenTime`
    `firstVisibleTime`
    `lastSeenTime`
    `lastVisibleTime`

Intent to implement in #1297

### CODE REVIEW

**Comment by avimehta** [comment]
@jridgewell The tests are failing with following error:

> Chrome 50.0.2661 (Linux 0.0.0) Visibility (tag: amp-analytics) fires for trivial config FAILED
>         TypeError: babelHelpers.defineProperty is not a function
>             at Visibility.listenOnce (/tmp/60a144f13bb7ba1df80a3d72d415a67d.browserify:56729:29 <- /usr/local/google/home/avimehta/amp/amphtml/src/service/visibility-impl.js:250:11)
>             at listen (/tmp/60a144f13bb7ba1df80a3d72d415a67d.browserify:81747:16 <- /usr/local/google/home/avimehta/amp/amphtml/test/functional/test-visibility.js:75:15)
>             at Context.<anonymous> (/tmp/60a144f13bb7ba1df80a3d72d415a67d.browserify:81759:5 <- /usr/local/google/home/avimehta/amp/amphtml/test/functional/test-visibility.js:87:4)
> Chrome 50.0.2661 (Linux 0.0.0) Visibility (tag: amp-analytics) fires for non-trivial config FAILED
>         TypeError: babelHelpers.defineProperty is not a function
>             at Visibility.listenOnce (/tmp/60a144f13bb7ba1df80a3d72d415a67d.browserify:56729:29 <- /usr/local/google/home/avimehta/amp/amphtml/src/service/visibility-impl.js:250:11)
>             at listen (/tmp/60a144f13bb7ba1df80a3d72d415a67d.browserify:81747:16 <- /usr/local/google/home/avimehta/amp/amphtml/test/functional/test-visibility.js:75:15)
>             at Context.<anonymous> (/tmp/60a144f13bb7ba1df80a3d72d415a67d.browserify:81764:5 <- /usr/local/google/home/avimehta/amp/amphtml/test/functional/test-visibility.js:92:4)
> Chrome 50.0.2661 (Linux 0.0.0) Visibility (tag: amp-analytics) fires only once FAILED
>         TypeError: babelHelpers.defineProperty is not a function
>             at Visibility.listenOnce (/tmp/60a144f13bb7ba1df80a3d72d415a67d.browserify:56729:29 <- /usr/local/google/home/avimehta/amp/amphtml/src/service/visibility-impl.js:250:11)

Are (http://es6-features.org/#ComputedPropertyNames)[computer property names] not supported in tests?

**Comment by jridgewell** [comment]
> Are (http://es6-features.org/#ComputedPropertyNames)[computer property names] not supported in tests?

Likely not, though we could easily add support.

**Comment by avimehta** [comment]
Can we do that? How do I go about doing it?

**Comment by jridgewell** [comment]
We define the babel helpers in `third_party/babel/custom-babel-helpers.js`. Looks like this should work, but you'll need to confirm:

```js
babelHelpers.defineProperty = function(obj, key, value) {
  obj[key] = value;
};

**Comment by avimehta** [comment]
ptal. 

/cc @cramforce

File: src/service/visibility-impl.js [hunk 1]
```diff
@@ -14,13 +14,153 @@
...
+/** @const {string} */
+const CONTINUOUS_TIME_MIN = 'continuousTimeMin';
+
+/** @const {string} */
+const TOTAL_TIME_MAX = 'totalTimeMax';
+
+/** @const {string} */
+const TOTAL_TIME_MIN = 'totalTimeMin';
+
+/** @const {string} */
```
Comment (cramforce) [inline src/service/visibility-impl.js]: No need for these annotations. Closure compiler infers this.

File: extensions/amp-analytics/0.1/instrumentation.js [hunk 2]
```diff
@@ -221,15 +222,15 @@ export class InstrumentationService {
    * @private
    */
   createVisibilityListener_(callback, config) {
-    if (!this.isVisibilitySpecValid_(config)) {
+    if (!isVisibilitySpecValid(config, this.isViewabilityExperimentOn_())) {
```
Comment (cramforce) [inline extensions/amp-analytics/0.1/instrumentation.js]: Why not make a simple condition like

```
if (this.isViewabilityExperimentOn_() {
  return;
}
```

**Comment by cramforce** [comment]
Looks good. Just a few comments.

**Comment by avimehta** [comment]
Thanks for the review. Merging the PR.

File: extensions/amp-analytics/0.1/instrumentation.js (hunk 2 above)
Comment (avimehta) [inline extensions/amp-analytics/0.1/instrumentation.js]: Changed this a bit.

File: src/service/visibility-impl.js (hunk 1 above)
Comment (avimehta) [inline src/service/visibility-impl.js]: Removed these

**Pull request merged**